File: Sources/SomnioCore/Models/CharacterClass.swift
import Foundation

public enum CharacterClass: Int16, Sendable, Equatable, Hashable, CaseIterable {
    case fighter = 0
    case lancer = 1
    case warrior = 2
    case thief = 3
    case hunter = 4
    case gangster = 5
    case cleric = 6
    case mage = 7

    /// The class display label resolved through the SomnioCore string catalog. The English
    /// source string doubles as the catalog key per Apple's `String(localized:)` convention;
    /// the German translations carry the original game's authoritative labels.
    public var displayName: String {
        CoreCatalog.localized(localizationKey)
    }

    private var localizationKey: String {
        switch self {
        case .fighter: return "Fighter"
        case .lancer: return "Lancer"
        case .warrior: return "Warrior"
        case .thief: return "Thief"
        case .hunter: return "Hunter"
        case .gangster: return "Gangster"
        case .cleric: return "Cleric"
        case .mage: return "Mage"
        }
    }
}

File: packages/core/src/characterClass.ts
/**
 * Named maps rather than inlined literals: these raw values go out on the wire in the `register`
 * payload as an opaque `Int16`, so a drifted table would keep sending the old numbers with the
 * round-trip and golden-frame suites both green. `enum-raws.test.ts` pins them to literals. The
 * display names resolve through `coreCatalog`, keyed by the English label.
 */
export const CHARACTER_CLASS = {
  fighter: 0,
  lancer: 1,
  warrior: 2,
  thief: 3,
  hunter: 4,
  gangster: 5,
  cleric: 6,
  mage: 7,
} as const
export type CharacterClass = (typeof CHARACTER_CLASS)[keyof typeof CHARACTER_CLASS]

export const GENDER = { male: 0, female: 1 } as const
export type Gender = (typeof GENDER)[keyof typeof GENDER]
